src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsSplitSelector.js

```diff
@@ -60,183 +60,157 @@ const defaultProps = {
-            const {recentReports, personalDetails, userToInvite} = OptionsListUtils.getNewChatOptions(
-                this.props.reports,
-                this.props.personalDetails,
-                this.props.betas,
-                isOptionInList ? prevState.searchTerm : '',
-                newSelectedOptions,
-                CONST.EXPENSIFY_EMAILS,
-            );
-            return {
-                recentReports,
-                personalDetails,
-                userToInvite,
-                searchTerm: isOptionInList ? prevState.searchTerm : '',
-            };
-        });
-    }
-
-    render() {
-        const maxParticipantsReached = this.props.participants.length === CONST.REPORT.MAXIMUM_PARTICIPANTS;
-        const sections = this.getSections(maxParticipantsReached);
-        const headerMessage = OptionsListUtils.getHeaderMessage(
-            this.state.personalDetails.length + this.state.recentReports.length !== 0,
-            Boolean(this.state.userToInvite),
-            this.state.searchTerm,
-            maxParticipantsReached,
+        const {recentReports, personalDetails, userToInvite} = OptionsListUtils.getNewChatOptions(
+            props.reports,
+            props.personalDetails,
+            props.betas,
+            isOptionInList ? searchTerm : '',
+            newSelectedOptions,
+            CONST.EXPENSIFY_EMAILS,
         );
-        const isOptionsDataReady = ReportUtils.isReportDataReady() && OptionsListUtils.isPersonalDetailsReady(this.props.personalDetails);
-
-        return (
-            <View style={[styles.flex1, styles.w100, this.props.participants.length > 0 ? this.props.safeAreaPaddingBottomStyle : {}]}>
-                <OptionsSelector
-                    canSelectMultipleOptions
-                    sections={sections}
-                    selectedOptions={this.props.participants}
-                    value={this.state.searchTerm}
-                    onSelectRow={this.toggleOption}
-                    onChangeText={this.updateOptionsWithSearchTerm}
-                    headerMessage={headerMessage}
-                    boldStyle
-                    shouldShowConfirmButton
-                    confirmButtonText={this.props.translate('common.next')}
-                    onConfirmSelection={this.finalizeParticipants}
-                    textInputLabel={this.props.translate('optionsSelector.nameEmailOrPhoneNumber')}
-                    safeAreaPaddingBottomStyle={this.props.safeAreaPaddingBottomStyle}
-                    shouldShowOptions={isOptionsDataReady}
-                />
-            </View>
+
+        setNewChatOptions({
+            recentReports,
+            personalDetails,
+            userToInvite,
+        });
+        props.onAddParticipants(newSelectedOptions);
+    };
+
+    /**
+     * Once a single or more users are selected, navigates to next step
+     */
+    const finalizeParticipants = () => {
+        props.onStepComplete();
+    };
```

Comment:
Hi, @allroundexperts 
There is no reason.
I paid more attention to convert toggleOption and getSections function because these functions are main functions in this component.
So I converted them first and then finalizeParticipants.
That's why changed the order.

Reply:
Just for my understanding, why don't we use `withCallback` here? I assume it's not necessary, but I'm still trying to understand hooks.

Reply:
Hi, @allroundexperts 
Updated to use the prop directly.